From ff33f536350b62c872e1893021273f0ec93094e3 Mon Sep 17 00:00:00 2001
From: Xubao Jiang <xubaoj@codeaurora.org>
Date: Mon, 26 Apr 2021 18:20:45 +0800
Subject: [PATCH] Disable close_range

close_range first appeared in Linux 5.9. The system call number __NR_close_range
defined in systemd and Linux is 436.

In linux-msm 5.4 close_range is not supported. And the system call nubmer 436 is
used by __NR_process_madvise. This causes the call of close_range returns invalid
argument from process_madvise.

So disable close_range here to fix boot up issue with linux-msm 5.4.

This patch would no longer be necessary if a newer kernel which supports close_range
is introduced.

Upstream-Status: Inappropriate [disable feature]

Signed-off-by: Xubao Jiang <xubaoj@codeaurora.org>
---
 src/basic/fd-util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/basic/fd-util.c b/src/basic/fd-util.c
index 07a7b3a..0addf6d 100644
--- a/src/basic/fd-util.c
+++ b/src/basic/fd-util.c
@@ -216,7 +216,7 @@ static int cmp_int(const int *a, const int *b) {
 }
 
 int close_all_fds(const int except[], size_t n_except) {
-        static bool have_close_range = true; /* Assume we live in the future */
+        static bool have_close_range = false;
         _cleanup_closedir_ DIR *d = NULL;
         struct dirent *de;
         int r = 0;
-- 
2.7.4

